File: ontology-engine/graph-engine/src/main/java/org/sunbird/graph/engine/actor/NodeManager.java
package org.sunbird.graph.engine.actor;

import org.sunbird.actor.router.ActorConfig;
import org.sunbird.common.dto.Request;
import org.sunbird.graph.engine.DataNode;
import org.sunbird.graph.mgr.BaseGraphManager;


@ActorConfig(tasks = {"createDataNode"})
public class NodeManager extends BaseGraphManager {

    // TODO: put it as part of request;
    String graphId = "domain";

    @Override
    public void onReceive(Request request) throws Throwable {
        String operation = request.getOperation();
        switch (operation) {
            case "createDataNode":
                createDataNode(request);
                break;
            default:
                ERROR(operation);
        }
    }

    private void createDataNode(Request request) throws Exception {
        DataNode node = new DataNode(this, graphId, request.getObjectType(), "1.0");
        node.create(request);
    }

}

File: ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/engine/actor/NodeManager.scala
package org.sunbird.graph.engine.actor

import org.sunbird.actor.router.ActorConfig
import org.sunbird.common.dto.{Request, Response}
import org.sunbird.graph.mgr.BaseGraphManager
import org.sunbird.graph.nodes.DataNode

@ActorConfig(tasks = Array("createDataNode", "saveExternalProperties"))
class NodeManager extends BaseGraphManager {

    val graphId = "domain"

    @throws[Throwable]
    override def onReceive(request: Request): Unit = {
        val operation = request.getOperation
        operation match {
            case "createDataNode" =>
                createDataNode(request)
            case "saveExternalProperties" =>
                sender() ! new Response()
            case _ =>
                ERROR(operation)
        }
    }

    @throws[Exception]
    private def createDataNode(request: Request): Unit = {
        val node = new DataNode(this, graphId, request.getObjectType, "1.0")
        request.getContext.put("graph_id", graphId)
        node.create(request)
    }
}
